lib/client-fund-contract.js

```diff
@@ -5,7 +5,9 @@ const clientFundDeployment = require('./abis/ClientFund');
 
 class ClientFundContract extends ethers.Contract {
     constructor(walletOrProvider) {
-        const clientFundAddress = clientFundDeployment.networks[walletOrProvider.chainId || walletOrProvider.provider.chainId].address;
+        const provider = walletOrProvider.provider ? walletOrProvider.provider : walletOrProvider;
+        const chainId = provider._network.chainId;
```

Comment:
Youre using a "private" member... good example of why "convention based" programming sucks!